builtin/docker/platform.go

```diff
@@ -126,65 +297,40 @@ func (p *Platform) Status(
 
-	// currently the docker platform only deploys 1 container
-	containerInfo, err := cli.ContainerInspect(ctx, deployment.Container)
-	if err != nil {
-		return nil, err
-	}
-
-	// Create our status report
-	var result sdk.StatusReport
-	result.External = true
-
-	// NOTE(briancain): The docker platform currently only deploys a single
-	// container, so for now the status report makes the same assumption.
-
-	log.Debug("querying docker for container health")
+	rm := p.resourceManager(log, nil)
 
-	resources := []*sdk.StatusReport_Resource{{
-		Name: containerInfo.Name,
-	}}
-
-	if containerInfo.State.Health != nil {
-		// Built-in Docker health reporting
-		// NOTE: this only works if the container has configured health checks
-
-		switch containerInfo.State.Health.Status {
-		case "Healthy":
-			resources[0].Health = sdk.StatusReport_READY
-			resources[0].HealthMessage = "container is running"
-		case "Unhealthy":
-			resources[0].Health = sdk.StatusReport_DOWN
-			resources[0].HealthMessage = "container is down"
-		case "Starting":
-			resources[0].Health = sdk.StatusReport_ALIVE
-			resources[0].HealthMessage = "container is starting"
-		default:
-			resources[0].Health = sdk.StatusReport_UNKNOWN
-			resources[0].HealthMessage = "unknown status reported by docker for container"
+	// If we don't have resource state, this state is from an older version
+	// and we need to manually recreate it.
+	if deployment.ResourceState == nil {
+		if err := rm.Resource("container").SetState(&Resource_Container{
+			Id: deployment.Container,
+		}); err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to manually set container resource state while restoring from an old deployment: %s", err)
+		}
+		if err := rm.Resource("network").SetState(&Resource_Network{
+			Name: "waypoint",
+		}); err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to manually set network resource state while restoring from an old deployment: %s", err)
 		}
 	} else {
-		// Waypoint container inspection
-
-		if containerInfo.State.Running && containerInfo.State.ExitCode == 0 {
-			resources[0].Health = sdk.StatusReport_READY
-			resources[0].HealthMessage = "container is running"
-		} else if containerInfo.State.Restarting || containerInfo.State.Status == "created" {
-			resources[0].Health = sdk.StatusReport_ALIVE
-			resources[0].HealthMessage = "container is still starting"
-		} else if containerInfo.State.Dead || containerInfo.State.OOMKilled || containerInfo.State.ExitCode != 0 {
-			resources[0].Health = sdk.StatusReport_DOWN
-			resources[0].HealthMessage = "container is down"
-		} else {
-			resources[0].Health = sdk.StatusReport_UNKNOWN
-			resources[0].HealthMessage = "unknown status for container"
+		// Load our set state
+		if err := rm.LoadState(deployment.ResourceState); err != nil {
+			return nil, status.Errorf(codes.Internal, "resource manager failed to load state: %s", err)
 		}
 	}
 
+	resources, err := rm.StatusAll(ctx, log, sg, cli, ui)
```

Comment:
There seems to be a lot of duplication in this function (`Status`) that I thought would be gone now with this. I wonder if we're missing one last glue piece here. We don't need to do it for this PR or even for 0.5 but I woudl've expected that you could just do something like:

`return rm.StatusAllReport()` or `return resources.StatusReport` or something to get the proper structure out.

Reply:
I could see adding a `rm.StatusReport` function that called `StatusAll` behind the scenes and returned a `resource.StatusReport` with `Resources`, `GeneratedTime`, and `External` filled in. 

I think the plugins would still need to figure out what the overall status report `Health` and `HealthMessage` is based on the combination of statuses though.